nixos/modules/services/misc/gitea.nix

```diff
@@ -277,7 +278,46 @@ in
   };
 
   config = mkIf cfg.enable {
-    services.postgresql.enable = mkIf usePostgresql (mkDefault true);
+    assertions = [
+      { assertion = cfg.database.createDatabase -> cfg.database.user == cfg.user;
+        message = "services.gitea.database.user must match services.gitea.user if the database is to be automatically provisioned";
```

Comment:
The `createDatabase` option provisions a local `mysql` or `postgresql` database using local socket authentication, which ties a local user/system account to a `mysql`/`postgresql` account.

Reply:
The `host` option isn't nullable, so we can't really check. As for making the `host` option nullable... if we do that it raises the question as to why `port`, `password`, `passwordFile`, `user`, and 'path` aren't nullable either as they won't be used.

I'm not sure I gave you a great answer but I think that is a problem to be solved in a separate PR anyways.

Reply:
After skimming through some other modules with database integration I guess you're right, this should be fixed in a different PR and is good enough now. Sorry for bringing that up here.

Reply:
@Ma27 I'm glad you brought it up. It isn't clear exactly how this problem should be solved and there are a couple options. I had previously discussed with @floki the possibility of `services.foo.database` being nullable and the null state would indicate the service should provision a database however the service see fit. I was convinced this might not be flexible enough as the user can no longer specify a database name.

If you have any thoughts I'd love to hear them in https://github.com/NixOS/nixpkgs/issues/62073